drivers/mbox/mbox_nrf_bellboard_local.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_bellboard_local
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+
+#include <haly/nrfy_bellboard.h>
+
+#define BELLBOARD_NUM_IRQS 4U
+
+BUILD_ASSERT(DT_NUM_IRQS(DT_DRV_INST(0)) <= BELLBOARD_NUM_IRQS,
+	     "# interrupt exceeds maximum");
+
+BUILD_ASSERT(DT_INST_PROP_LEN(0, nordic_interrupt_mapping) == 2 * DT_NUM_IRQS(DT_DRV_INST(0)),
+	     "# interrupt mappings != # interrupts");
+
+/* BELLBOARD IRQ list */
+#define BELLBOARD_IRQ_GET_OR_ZERO(irq_)                                                            \
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(0, irq_), (DT_INST_IRQ_BY_NAME(0, irq_, irq)), (0U))
+
+static const uint8_t irqs[BELLBOARD_NUM_IRQS] = {
+	BELLBOARD_IRQ_GET_OR_ZERO(irq0),
+	BELLBOARD_IRQ_GET_OR_ZERO(irq1),
+	BELLBOARD_IRQ_GET_OR_ZERO(irq2),
+	BELLBOARD_IRQ_GET_OR_ZERO(irq3),
+};
+
+/* BELLBOARD IRQ mappings */
+#define BELLBOARD_GET_IRQ_MAPPING_OR_ZERO(idx)                                                     \
+	COND_CODE_1(DT_INST_PROP_HAS_IDX(0, nordic_interrupt_mapping, idx),                        \
+		    (DT_INST_PROP_BY_IDX(0, nordic_interrupt_mapping, idx)), (0U))
+
+static const uint32_t irq_mappings[BELLBOARD_NUM_IRQS] = {
+	BELLBOARD_GET_IRQ_MAPPING_OR_ZERO(0),
+	BELLBOARD_GET_IRQ_MAPPING_OR_ZERO(2),
+	BELLBOARD_GET_IRQ_MAPPING_OR_ZERO(4),
+	BELLBOARD_GET_IRQ_MAPPING_OR_ZERO(6),
```

Comment:
I think it's not going to work as intended. For instance, if you use the example from binding:
```
      interrupts = <98 NRF_DEFAULT_IRQ_PRIORITY>,
                   <99 NRF_DEFAULT_IRQ_PRIORITY>;
      interrupt-names = "irq2", "irq3";
      nordic,interrupt-mapping = <0x0000000f 2>, <0x000000f0 3>;
```
you'll get: `irqs = { 0, 0, 98, 99 };` and `irq_mappings = { 0x0000000f, 0x000000f0, 0, 0 };`

Reply:
How about using designated initializers and `LISTIFY` to fill `irq_mappings`?

Reply:
good catch, please check current revision

Reply:
> good catch, please check current revision

It still won't handle properly the above example. I think something like this should be used:
```
#define EVT_MAPPING_ITEM(idx) DT_INST_PROP_BY_IDX(0, nordic_interrupt_mapping, idx)
#define BELLBOARD_GET_EVT_MAPPING(idx, _)				\
	COND_CODE_1(DT_INST_PROP_HAS_IDX(0, nordic_interrupt_mapping,	\
					 UTIL_INC(UTIL_X2(idx))),	\
		([EVT_MAPPING_ITEM(UTIL_INC(UTIL_X2(idx)))] =		\
			EVT_MAPPING_ITEM(UTIL_X2(idx)),),		\
		())

static const uint32_t evt_mappings[BELLBOARD_NUM_IRQS] = {
	LISTIFY(DT_NUM_IRQS(DT_DRV_INST(0)), BELLBOARD_GET_EVT_MAPPING, ())
};
```
Then it won't be also needed to require that "# interrupt mappings == # interrupts";

Reply:
you're right, updated the driver to user you sample, thanks